mycalculator/src/components/Calculator.js

```diff
@@ -0,0 +1,199 @@
+            clearBtn:"AC"
+        }): this.setState({
+            displayValue: 0,
+            operation:null,
+            previousValue:0
+        })
+    }
+
+    handlePercentageClick = () =>{
+        let {displayValue}=this.state
+        this.setState({
+            displayValue:displayValue/100
+        })
+    }
+
+    handleNegPotClick = ()=>{
+        let {displayValue}=this.state
+        this.setState({
+            displayValue:-displayValue
+        })
+    }
+
+    handleOperationClick = (e)=>{
+        let {name} = e.target
+        let {displayValue}=this.state
+        // debugger
+        this.setState({
+            waitingForNewValue:true,
+            operation:name,
+            previousValue:Number(displayValue),
+        })
+    }
+
+    handleEqualOperationClick = ()=>{
+        let {displayValue,operation,previousValue}=this.state;
+      console.log(displayValue.length)
+        if(operation==="+"){
+            if(displayValue.length<=9){
+                this.setState({
+                    displayValue:(Number(displayValue)+Number(previousValue))
+                })
+            }else{
+                this.setState({
+                    displayValue:(Number(displayValue)+Number(previousValue)).toPrecision(4)
+                })
+            }
+        }
+        else if(operation==="-"){
+            if(displayValue.length<=9){
+                this.setState({
+                    displayValue:(Number(previousValue) - Number(displayValue))
+                })
+            }else{
+                this.setState({
+                    displayValue:(Number(previousValue) - Number(displayValue)).toPrecision(4)
+                })
+            }
+        }
+        else if(operation==="x"){
+            if(displayValue.length<=9){
+                this.setState({
+                    displayValue:(Number(previousValue) * Number(displayValue))
+                })
+
+            }else{
+                this.setState({
+                    displayValue:(Number(previousValue) * Number(displayValue)).toPrecision(4)
+                })
+            }
+        }
+        else{
+            if(displayValue.length<=9){
+                this.setState({
+                    displayValue:(Number(previousValue) / Number(displayValue))
+                })
+            }else{
+                this.setState({
+                    displayValue:(Number(previousValue) / Number(displayValue)).toPrecision(4)
+                })
+            }
+        }
+    }
+
+
+    render(){
+        console.log(this.state);
+        let {displayValue,clearBtn}=this.state;
+        return( 
+            <div className="display">
+            <div className="calculator">
+            <div className="col-12 inputview">{displayValue}</div>
+            <button className="button col-3" name="AC" onClick={(e)=>{
+                this.handleACClick()}}>{clearBtn}</button>
+
+            <button className="button col-3" name="%" onClick={(e)=>{
```

Comment:
All these buttons could have been one component that is imported to your calculator. Or even better, two components where one loops through your numbers/ operations and uses a special`<Button />` component that adds listeners to each button for your instead of doing it by hand.